torchaudio/datasets/esc.py

```diff
@@ -0,0 +1,185 @@
+                 folds: tuple = FOLDS,
+                 download: bool = False,
+                 transform: Module = None) -> None:
+
+        super().__init__()
+
+        self.root = root
+        self.required_folds = folds
+        self.transform = transform
+
+        self.url = URL["esc-50"]
+        self.nb_class = 50
+        self.target_directory = os.path.join(self.root, FOLDER_IN_ARCHIVE)
+
+        # Dataset must exist to continue
+        if download:
+            self.download()
+        # elif not self.check_integrity(self.target_directory):
+        #     raise RuntimeError("Dataset not found or corrupted. \n\
+        #         You can use download=True to download it.")
+
+        # Prepare the medata
+        self._filenames = []
+        self._folds = []
+        self._targets = []
+        self._esc10s = []
+        self._load_metadata()
+
+    def __getitem__(self, index: int) -> Tuple[Tensor, int]:
+        """
+        Args:
+            index (int): Index
+
+        Returns:
+            tuple: (raw_audio, sr, target).
+        """
+        data, sampling_rate, target = self.load_item(index)
+
+        if self.transform is not None:
+            data = self.transform(data)
+            data = data.squeeze()
+
+        return data, sampling_rate, target
+
+    def __len__(self) -> int:
+        return len(self._filenames)
+
+    def _load_metadata(self) -> None:
+        """Read the metadata csv file and gather the information needed."""
+        # HEADER COLUMN NUMBER
+        c_filename = 0
+        c_fold = 1
+        c_target = 2
+        c_esc10 = 4
+
+        # Read the csv file and remove header
+        path = os.path.join(self.target_directory, META_FOLDER, "esc50.csv")
+        with open(path, "r") as fp:
+            data = fp.read().splitlines()[1:]
+
+            for line in data:
+                items = line.split(",")
+
+                self._filenames.append(items[c_filename])
+                self._folds.append(int(items[c_fold]))
+                self._targets.append(int(items[c_target]))
+                self._esc10s.append(eval(items[c_esc10]))
+
+        self._filenames = numpy.asarray(self._filenames)
+        self._folds = numpy.asarray(self._folds)
+        self._targets = numpy.asarray(self._targets)
+        self._esc10s = numpy.asarray(self._esc10s)
+
+        # Keep only the required folds
+        folds_mask = sum([self._folds == f for f in self.required_folds]) >= 1
+
+        self._filenames = self._filenames[folds_mask]
+        self._targets = self._targets[folds_mask]
+        self._esc10s = self._esc10s[folds_mask]
+
+    def download(self) -> None:
+        """Download the dataset and extract the archive"""
+        if self.check_integrity(self.target_directory):
+            print("Dataset already downloaded and verified.")
+
+        else:
+            archive_path = os.path.join(self.root, FOLDER_IN_ARCHIVE + ".zip")
+
+            download_url(self.url, self.root)
```

Comment:
nit: (1) we currently download the file only if it doesn't exist, (2) we offer the checksum of the archive